src/components/DataProvider/index.js

```diff
@@ -13,11 +14,25 @@ const calculateDomainFromData = (data, accessor) => {
   return [extent[0] - diff * 0.025, extent[1] + diff * 0.025];
 };
 
+const deleteUndefinedFromObject = obj => {
```

Comment:
Based on the name, this sounds like it should be calling `delete obj[k]` directly, not returning a copy. I don't have a suggestion for a better name (or that this should be the behavior), but it just stood out to me.